docstore/awsdynamodb/urls.go

```diff
@@ -115,25 +107,21 @@ func (o *URLOpener) processURL(u *url.URL) (db *dyn.DynamoDB, tableName, partiti
 		return nil, "", "", "", nil, fmt.Errorf("open collection %s: URL path must be empty, only the host is needed", u)
 	}
 
-	configProvider := &gcaws.ConfigOverrider{
-		Base: o.ConfigProvider,
-	}
-	overrideCfg, err := gcaws.ConfigFromURLParams(q)
+	cfg, err := gcaws.V2ConfigFromURLParams(context.Background(), q)
 	if err != nil {
 		return nil, "", "", "", nil, fmt.Errorf("open collection %s: %v", u, err)
 	}
-	configProvider.Configs = append(configProvider.Configs, overrideCfg)
-	db, err = Dial(configProvider)
+	db, err = Dial(cfg)
 	if err != nil {
 		return nil, "", "", "", nil, fmt.Errorf("open collection %s: %v", u, err)
 	}
 	return db, tableName, partitionKey, sortKey, opts, nil
 }
 
 // Dial gets an AWS DynamoDB service client.
-func Dial(p client.ConfigProvider) (*dyn.DynamoDB, error) {
-	if p == nil {
+func Dial(p aws.Config) (*dyn.Client, error) {
```

Comment:
This is a breaking change.

Reply:
This has changed to the equivalent type used for AWS SDK V2, again, please see the comment below on why that change has already been discussed and is necessary.